pyproject.toml

```diff
@@ -1,3 +1,10 @@
+[project]
+name = "earthaccess"
+version = "0.8.1"
+
+[tool.setuptools]
+packages = ["earthaccess"]
+
```

Comment:
The changes in this file were needed to make `pip install -e .` work. Open to suggestions from others on a better way to do this (this was me just working through errors until things worked)

Reply:
I don't think we need setuptools *at all* and could/should just remove `setup.py`. Notably, it states:
```
# This is a shim to hopefully allow Github to detect the package, build is done with poetry
```
https://github.com/nsidc/earthaccess/blob/main/setup.py

It was added in #181 by @betolink  -- Luis was this just so GitHub can recognise [who depends on it](https://github.com/nsidc/earthaccess/network/dependents)?

Reply:
Okay, if I checkout this branch, remove these changes to `pyproject.toml` (other than the deps update), and remove `setup.py`, this is the error message from pip:
```
ERROR: Project file://.../earthdata has a 'pyproject.toml' and its build backend is missing the 'build_editable' hook. Since it does not have a 'setup.py' nor a 'setup.cfg', it cannot be installed in editable mode. Consider using a build backend that supports PEP 660.
```

If I change the build system to:
```yaml
[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```
as recommended here: https://python-poetry.org/docs/pyproject#poetry-and-pep-517

it works

Reply:
Looks good!